qcfractal/storage_sockets/sqlalchemy_socket.py

```diff
@@ -1135,33 +1135,80 @@ def add_results(self, record_list: List[ResultRecord]):
             Dict with keys: data, meta
             Data is the ids of the inserted/updated/existing docs
         """
-
         meta = add_metadata_template()
 
-        result_ids = []
-        with self.session_scope() as session:
-            for result in record_list:
+        result_ids = ["placeholder"] * len(record_list)
+
+        results_list = []
+        existing_res = {}
+        new_record_idx, duplicates_idx = [], []
+        conds = []
+
+        conds = [
+            and_(
+                ResultORM.program == res.program,
+                ResultORM.driver == res.driver,
+                ResultORM.method == res.method,
+                ResultORM.basis == res.basis,
+                ResultORM.keywords == res.keywords,
+                ResultORM.molecule == res.molecule,
+            )
+            for res in results_list
+        ]
 
-                doc = session.query(ResultORM).filter_by(
-                    program=result.program,
-                    driver=result.driver,
-                    method=result.method,
-                    basis=result.basis,
-                    keywords=result.keywords,
-                    molecule=result.molecule,
+        with self.session_scope() as session:
+            docs = session.query(ResultORM).filter(or_(*conds)).all()
```

Comment:
Please change this to only return the fields you are interested in to reduce the cost of the query (the six tuple of interest + the id).

Reply:
Done in the new commit.